f3read.c

```diff
@@ -436,6 +436,12 @@ int main(int argc, char **argv)
 
 	restrict_fsaccess(&args.dev_path);
 
+#ifdef __OpenBSD__
+	assert(!unveil(args.dev_path, "r"));
+	assert(!unveil(NULL, NULL));
+	assert(!pledge("stdio rpath", ""));
+#endif
```

Comment:
Add a second `bool write` parameter to `restrict_fsaccess()` and move these lines to `restrict_fsaccess()`.

Reply:
unveil(2)? Yes, definitely. pledge(2) is about what syscalls can you do, not what parts of the FS is your access restricted to. I'd like to keep it in main(), there it also serves documentation purposes. (AFAIK more complex apps do even further different pledge(2)s in different treads.)

Reply:
I'm sorry, I cannot accept these functions in `main()`. I don't have much time to maintain F3, and if I start to accept code specific to a single platform on the central part of F3, the accumulation of them will end up adding a toll on my time.

Reply:
OK, let’s just not include platform-specific code in this repo at all. I'll maintain both pledge(2) and unveil(2) in the upcoming OpenBSD port.